lib/diagnostics_win.h

```diff
@@ -27,6 +27,7 @@ typedef LONG       NTSTATUS;
 typedef LONG       KPRIORITY;
 
 //MinGW-W64 defines this struct in its own header
+#define HAVE_CLIENT_ID 1
```

Comment:
I'm not sure that this macro is good because it can be the case when structs will be indefined.
Maybe we need something like this?
#if !defined(__MINGW32__) and !defined(HAVE_VM_COUNTERS)